fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -3051,19 +3099,38 @@ public Bundle doHistory(MultivaluedMap<String, String> queryParameters, String r
 
             StringBuilder nextRequest = new StringBuilder();
             nextRequest.append(serviceBase);
+            if (resourceType != null) {
+                // note that the serviceBase includes /_history for whole system history,
+                // but not for the resourceType/ whole history interaction
+                nextRequest.append("/").append(resourceType).append("/_history");
+            }
             nextRequest.append("?");
             nextRequest.append("_count=").append(count);
+            
+            if (resourceType == null && historyContext.getResourceTypes().size() > 0) {
+                nextRequest.append("&_type=");
+                nextRequest.append(String.join(",", historyContext.getResourceTypes()));
+            }
+            
+            if (historyContext.isExcludeTransactionTimeoutWindow()) {
+                nextRequest.append("&_excludeTransactionTimeoutWindow=true");
+            }
 
-            if (historyContext.getSince() != null && historyContext.getSince().getValue() != null) {
-                // As _since was given, we need to go with time-based paging, and the client
-                // will be responsible for managing duplicates.
+            switch (historyContext.getHistorySortOrder()) {
+            case ASC_LAST_UPDATED:
+                nextRequest.append("&_sort=" + HistorySortOrder.ASC_LAST_UPDATED.toString());
                 nextRequest.append("&_since=").append(lastChangeTime.atZone(UTC).format(DateTime.PARSER_FORMATTER));
-            } else {
-                // in all other cases we fetch ordered by resource_id, and so the client
-                // should get the next page using the _afterHistoryId marker, allowing
-                // easy paging through the stream of changes without duplicates
+                break;
+            case DESC_LAST_UPDATED:
+                nextRequest.append("&_sort=" + HistorySortOrder.DESC_LAST_UPDATED.toString());
+                nextRequest.append("&_since=").append(lastChangeTime.atZone(UTC).format(DateTime.PARSER_FORMATTER));
```

Comment:
if we introduce a `_before` parameter, we could use that here instead of changing the meaning of `_since`

Reply:
My struggle exactly (from the earlier conversation). I think this makes sense. It'll need good documentation, especially when interacting with the transaction timeout window.

Reply:
Changed the semantics so that both _since and _before can be used regardless of sort direction.